packages/webdriverio/src/commands/browser/url.js

```diff
@@ -35,9 +37,29 @@
 import nodeUrl from 'url'
 import { validateUrl } from '../../utils'
 
-export default function url (path) {
+export default async function url (path, { inject } = {} ) {
+    /**
+     * If inject is a function, then we attach puppeteer to the existing browser instance and
+     * injects the passed function on the active tab
+     *
+     */
+    if (inject) {
+        if (typeof inject !== 'function') {
+            throw new Error('Parameter "inject" for url command needs to be type of function')
+        }
+        const browser = await this.getPuppeteer()
+        const allPages = await browser.pages()
+        for (let page of allPages) {
+            const state = await page.evaluate(() => document.visibilityState) // eslint-disable-line
+            if (state === 'visible') {
+                await page.evaluate(inject)
+                break
+            }
+        }
```

Comment:
1. let's make sure to only inject it in the page that is currently active
2. We need to use this here: https://pptr.dev/#?product=Puppeteer&version=v5.4.1&show=api-pageevaluateonnewdocumentpagefunction-args in order to have the script executed before the page loads